eip-0004.md

```diff
@@ -0,0 +1,39 @@
+# Ergo token standard
+
+* Author: catena2w,kushti,oskin1
+* Status: Proposed
+* Created: 07-Oct-2019
+* License: CC0
+* Forking: not needed 
+
+## Motivation 
+
+This standard provides a uniform way to issue Ergo tokens.
+A standard interface allows any tokens on Ergo to be re-used by other applications: from wallets to decentralized exchanges.
+
+
+## Ergo tokens background
+
+Ergo supports custom tokens as first-class citizens.
+Namely, a special register (R2) of a box contains (tokenId -> amount) pairs.
+A box can contain at most 255 of secondary tokens. However, there are also indirect limits: a box could be no more than 4 kilobytes, and also tokens add the computational cost of the transaction.
+
+A transaction can create out-of-thin-air tokens in its outputs if the token identifier is equal to the identifier of the first input box of the transaction.
+As the box identifier is cryptographically unique, there's no chance to have the second token with the same identifier while the hash function being used is collision-resistant.
+This rule also means that only one new token per transaction can be created.
+
+Unlike ergs, other tokens can be burnt: the total amount for a token in transaction inputs should be no less than the amount of the outputs.
+Storage rent component allows a miner to collect (or burn) all the tokens inside a box if it is expired, and there are not enough Ergs in the box to pay the storage rent fee.
+
+## Ergo tokens standard
+
+Though this is not required by the protocol, we propose the following structure for the box that issues a token:
+
+| Register        | Description                                     | Example                      |
+| --------------- |:-----------------------------------------------:| ----------------------------:|
+| R2              | Token id and amount pair                        | [("7d...09", 100000)]        |
+| R4              | Token verbose name (UTF-8 representation)       | "USD"                        |
```

Comment:
It is worth mentioning that the strings are actually byte arrays or `Coll[Byte]` type of ErgoScript, with bytes obtained using the corresponding encoder (UTF-8 in this case).

Reply:
@aslesarenko fixed.
Also example of a real such transaction in mainnet was added